bodhi/server/models.py

```diff
@@ -1817,7 +1817,7 @@ def _composite_karma(self):
         negative_karma = 0
         users_counted = set()
         for comment in self.comments_since_karma_reset:
-            if comment.karma and not comment.anonymous and comment.user.name not in users_counted:
+            if comment.karma and comment.user.name not in users_counted:
```

Comment:
We are going to need to take some action to ensure that historical anonymous karma does not get counted as karma once this patch is deployed. One way I can think to do that is to make a data migration that finds all anonymous comments, sets their karma field to 0, and modifies the comment text to indicate whether they had put a +1 or -1.

What do you think?

Reply:
I didn't realize that in the past(before 3.12, so 17 Dec 2018) anno users where allowed to leave karma(and critpathkarma, but I assume this field will be removed in 4.0). So removing karma from anno comments is mandatory hiere. But I'm not sure about leaving +1/-1 note about karma in text. For instance: [FEDORA-EPEL-2018-65bdd9ec8c](https://bodhi.fedoraproject.org/updates/FEDORA-EPEL-2018-65bdd9ec8c) shows that karma is +2(right panel), but actualy there are a few more votes for this update, and user can be suprised that there is a special algorithm for counting karma in backend. Maybe 4.0 will be a good opportunity to prepare some clarification in this topic, something like "Anonymous user karma:1 (anonymous votes are not take in to account when counting overall update karma)" or we can do some support for legacy "anno" comments leaving this kind of note below all anno comments in frontend(and maybe in CLI).

Reply:
Since we are removing the feature to add anonymous comments, I don't think we need to worry too much about how we store the historical information. I think it'd be OK to just modify the comment, and add ```"\n\nkarma: +1"``` to it to record what the karma had been.

Alternatively, we could treat the "anonymous" username as special since it historically was used to do this, and modify our karma counting algorithm to ignore karma from that particular username, instead of looking for the anonymous=True boolean.

Alternatively, we could just keep the boolean for now, and solve this problem later. We would just remove the ability for new updates to get the boolean set to True.

I'm OK with any of the above solutions.

I think our users are probably mostly aware that anonymous karma isn't counted. In any case, since we are removing anonymous comments, I think it's OK if we don't try to make it clear that anonymous comments are ignored for karma purposes, since only historic data will have them anymore.